apps/src/code-studio/initApp/project.js

```diff
@@ -999,15 +1005,16 @@ var projects = module.exports = {
     this.save().then(callback);
   },
   /**
-   * Freezes and saves the project. Also hides so that it's not available for deleting/renaming in the user's project list.
+   * Freezes the project. Also hides so that it's not available for
+   * deleting/renaming in the user's project list.
    */
   freeze(callback) {
+    if (!(current && current.isOwner)) {
+      return;
+    }
     current.frozen = true;
     current.hidden = true;
-    this.save().then(data => {
-      executeCallback(callback, data);
-      redirectEditView();
```

Comment:
we don't need to redirect here because we already reload the page in the callback: https://github.com/code-dot-org/code-dot-org/blob/07a34a27065ab491d6db5c9d40111871d740367a/apps/src/code-studio/showProjectAdmin.js#L18 if we are on an edit link, then the on reload we'll be redirected to the view link here: https://github.com/code-dot-org/code-dot-org/blob/8c96f6d14be0201ac3f9c0e6141075f6eb2014fe/apps/src/code-studio/initApp/project.js#L1390